lib/github_service/commands/cross_repo_test.rb

```diff
@@ -134,18 +137,37 @@ def parse_value(value)
         @value = value
 
         @test_repos, @repos = value.split(/\s+including\s+/)
-                                   .map { |repo_list| repo_list.split(",") }
+                                   .map { |repo_list| repo_list.split(",").map(&:strip) }
 
         # Add the identifier for the PR for this comment to @repos here
         @repos ||= []
         @repos  << "#{issue.repo_name}##{issue.number}"
 
+        repo_groups, @test_repos = @test_repos.partition { |repo_name| repo_name.start_with?("/") }
+        @test_repos += expand_repo_groups(repo_groups)
+
         @test_repos.map! { |repo_name| normalize_repo_name(repo_name.strip) }
         @repos.map!      { |repo_name| normalize_repo_name(repo_name.strip) }
 
         [@repos, @test_repos].each(&:uniq!)
       end
 
+      def expand_repo_groups(repo_groups)
+        # Expand out each group and prioritize any repos which were passed explicitly
+        # in the test_repos list by the user
+        #
+        # E.g. `cross-repo manageiq-providers-amazon#1234 /providers` should only
+        # include manageiq-providers-amazon#1234
+        repo_groups.flat_map { |repo_group| expand_repo_group(repo_group) }
+                   .reject { |repo_name| @test_repos.any? { |test_repo| test_repo.include?(repo_name) } }
```

Comment:
I think you can remove the reject and leave this as a pure function.  Then up above do

```ruby
@test_repos = @test_repos | expand_repo_groups(repo_groups)
```

which will do a set union and eliminate the dups for you.

Reply:
So I used `.include?` instead of a straight `|=` or `.uniq` because most likely a test_repo passed in by the user is going to be a PR/branch/sha e.g. `@miq-bot cross_repo_tests manageiq-providers-amazon#1234 /providers` (see the spec test covering this case) and we want those to take precedence over the expanded repo groups

Reply:
Oh I see...hmm then maybe we need a bit of an expanded comment as I didn't realize that was the problem you were trying to solve.  In any case, :+1:

Reply:
I can add that we're looking for a substring in the test_repos list specifically because we want to prioritize a PR over the bare repo otherwise I think the comment covers what we're doing and not necessarily how it is being done

Reply:
Actually I think I need to do a regex not just `.include?` because `manageiq-providers-azure` would also match on `manageiq-providers-azure_stack`

Reply:
Okay this should pick up on `(ManageIQ/)manageiq-providers-azure(#|@)1234` but not `manageiq-providers-azure_stack` now, spec added for this case also